ddl/ddl_worker_test.go

```diff
@@ -245,13 +246,16 @@ func doDDLJobErr(c *C, schemaID, tableID int64, tp model.ActionType, args []inte
 func checkCancelState(txn kv.Transaction, job *model.Job, test *testCancelJob) error {
 	var checkErr error
 	if test.cancelState == job.SchemaState {
-		errs, err := inspectkv.CancelJobs(txn, test.jobIDs)
-		if err != test.cancelRetErr {
-			checkErr = errors.Trace(err)
-		}
-		// It only tests cancel one DDL job.
-		if errs[0] != test.cancelRetErrs[0] {
-			checkErr = errors.Trace(errs[0])
+		if test.act == model.ActionAddIndex && job.SchemaState == model.StateWriteReorganization && job.SnapshotVer == 0 {
```

Comment:
This condition is too long, need to define a variable like
```
addIndexFirstReorg := test.act == model.ActionAddIndex && job.SchemaState == model.StateWriteReorganization && job.SnapshotVer == 0
```
Add add comment about why ignore this state.